### DISCUSSION TITLE
Ability to set REPO:TAG when creating image from Dockerfile

### DISCUSSION THREAD

**Discussion by igribkov** [testcontainers/testcontainers-go#1307] [General]
I'd like to say sorry if this question already has been asked, but is it possible to set an REPO:TAG for image created from Dockerfile?

as i can see this ability has not been implemented, and now REPO:TAG is UUID:UUID

```
func (p *DockerProvider) BuildImage(ctx context.Context, img ImageBuildInfo) (string, error) {
	repo := uuid.New()
	tag := uuid.New()

	repoTag := fmt.Sprintf("%s:%s", repo, tag)

	buildContext, err := img.GetContext()
	if err != nil {
		return "", err
	}

	buildOptions := types.ImageBuildOptions{
		BuildArgs:   img.GetBuildArgs(),
		Dockerfile:  img.GetDockerfile(),
		AuthConfigs: img.GetAuthConfigs(),
		Context:     buildContext,
		Tags:        []string{repoTag},
		Remove:      true,
		ForceRemove: true,
	}
```
so is it possible to add ability to pass REPO:TAG to BuildImage function?

**Comment by mdelapenya** [comment]
Hi @igribkov I think this is a really great feature request, I think doable! At the same time, I can see that having a random `UUID:UUID` could be desired for testing, where the images are ephemeral and probably destroyed after the test session finishes, but I'd like to understand more on your use case: in which cases you want to stand to a fixed `image:tag`?

If possible, would you consider contributing a PR with it? 🙏

**Reply by igribkov** [reply]
Hi @mdelapenya, sorry for so late response
Just for example i would like to understand from ```docker ps``` which container version has been started
```
CONTAINER ID   IMAGE                                                                       COMMAND                  CREATED          STATUS          PORTS                                                     NAMES
7ff1577c158d   dff89f2f-8eb4-4b9a-8982-d6fe5b92fd9d:6e83719e-171c-4d64-8a99-0893d2c733c8   "top -b"                 4 seconds ago    Up 3 seconds                                                              TEST_FROM_DOCKERFILE
```
I've understand that I have NAME, but in some cases I would like to get additional info from IMAGE

also by some reasons i would like to create a container with specified ```image:tag``` and use it in separate service

**Reply by mdelapenya** [reply]
Not sure why I added the wrong mention... 🤦 

Anyway, as I said above, this is a very valid use case. We can create an issue from this discussion and start a PR for it.

**Reply by igribkov** [reply]
@mdelapenya i've already created an Enhancement issue and finished with code, what should i do next?

**Reply by igribkov** [reply]
PR: https://github.com/testcontainers/testcontainers-go/pull/1508

**Comment by mdelapenya** [comment]
I think we can close this as implemented. Thanks!
